### PR TITLE
[RLlib] Incorporate VectorEncoder into PPORLModule and tests

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

This PR matches @smorad 's VectorEncoder to the current PPORLModule code.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/ppo/torch/ppo_torch_rl_module.py [hunk 1]
```diff
@@ -40,59 +41,25 @@ def get_ppo_loss(fwd_in, fwd_out):
...
         shared_encoder: Whether to share the encoder between the pi and value
     """
 
-    pi_config: FCConfig = None
-    vf_config: FCConfig = None
-    encoder_config: FCConfig = None
+    pi_config: ModelConfig = None
+    vf_config: ModelConfig = None
+    pi_encoder_config: ModelConfig = None
+    vf_encoder_config: ModelConfig = None
```
Comment (smorad) [inline rllib/algorithms/ppo/torch/ppo_torch_rl_module.py:61]: nit: Maybe these should be `EncoderConfig`?

File: rllib/algorithms/ppo/torch/ppo_torch_rl_module.py [hunk 2]
```diff
@@ -108,28 +75,27 @@ def setup(self) -> None:
...
-            self.encoder_pi = self.config.encoder_config.build()
-            self.encoder_vf = self.config.encoder_config.build()
-
-        self.pi = FCNet(
-            input_dim=self.config.pi_config.input_dim,
-            output_dim=self.config.pi_config.output_dim,
-            hidden_layers=self.config.pi_config.hidden_layers,
-            activation=self.config.pi_config.activation,
+        obs_spec = ModelSpec(
+            {  # bxt is just a name for better readability to indicated padded batch
```
Comment (smorad) [inline rllib/algorithms/ppo/torch/ppo_torch_rl_module.py:79]: 👍

File: rllib/algorithms/ppo/torch/ppo_torch_rl_module.py [hunk 3]
```diff
@@ -108,28 +75,27 @@ def setup(self) -> None:
...
-        self.vf = FCNet(
-            input_dim=self.config.vf_config.input_dim,
-            output_dim=self.config.vf_config.output_dim,
-            hidden_layers=self.config.vf_config.hidden_layers,
-            activation=self.config.vf_config.activation,
+        self.shared_encoder = self.config.shared_encoder_config.build(
+            input_spec=obs_spec
+        )
+        self.pi_encoder = self.config.pi_encoder_config.build(
+            self.shared_encoder.output_spec
```
Comment (smorad) [inline rllib/algorithms/ppo/torch/ppo_torch_rl_module.py:90]: You might want to call `spec.filter` here, not sure.

File: rllib/core/rl_module/encoder.py [hunk 4]
```diff
@@ -13,6 +13,9 @@
 
 # TODO (Kourosh): Find a better / more straight fwd approach for sub-components
 
+ENCODER_OUT = "embedding"
+STATE_IN = "state_in"
```
Comment (smorad) [inline rllib/core/rl_module/encoder.py:17]: `state` is super overloaded. Is it possible to call these `recurrent_state` or something?

**Comment by ArturNiederfahrenhorst** [comment]
Closing in favour of https://github.com/ray-project/ray/pull/31469

**Pull request closed without merging**